### PR TITLE
Storage entry template to reduce boilerplate

### PR DESCRIPTION
**Pull request by stavros-k**
Adds a template for the storage entry to use in apps, this will allow to easily add extra fields in the future, with only adapting the questions on the apps

Also bumps CI deps

---

New structure for hostPath and ixVolume

Also readOnly is added as option for all storage types (hostPath, ixVolume, smb).
This option is added only for additionalStorages.

```yaml
appStorage:
  additionalStorages:
   # hosPath ACL true
    - type: hostPath
      readOnly: true
      mountPath: /data1
      hostPathConfig:
        aclEnable: true
        acl:
          path: /mnt/pool/data1
  
    # hosPath ACL false
    - type: hostPath
      readOnly: true
      mountPath: /data2
      hostPathConfig:
        aclEnable: false
        hostPath: /mnt/pool/data2
  
    # ixVol ACL true
    - type: ixVolume
      mountPath: /data3
      ixVolumeConfig:
        datasetName: data3
        aclEnable: true
  
    # ixVol ACL false
    - type: ixVolume
      mountPath: /data4
      ixVolumeConfig:
        datasetName: data4
        aclEnable: false

    # SMB (no acl here)
    - type: smb-pv-pvc
      mountPath: /data5
      smbConfig:
        server: 10.0.0.1
        share: share-name
        domain: domain
        username: username
        password: password
        size: 1
```

### CODE REVIEW

File: .github/ct-install-config/common-ct-lint.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 remote: origin
 target-branch: master
-helm-extra-args: --timeout 600s --debug
+helm-extra-args: --debug
```
Comment (sonicaj) [inline .github/ct-install-config/common-ct-lint.yaml:3]: Curious, not required anymore ?

File: library/common/templates/app_functions/_postgres.tpl [hunk 2]
```diff
@@ -203,9 +201,7 @@ postgresdata:
         mountPath: /mnt/directories/postgres_data
 postgresbackup:
   enabled: true
-  type: {{ $backup.type }}
-  datasetName: {{ $backup.datasetName | default "" }}
-  hostPath: {{ $backup.hostPath | default "" }}
+  {{- include "ix.v1.common.app.storageOptions" (dict "storage" $backup) | nindent 2 }}
```
Comment (sonicaj) [inline library/common/templates/app_functions/_postgres.tpl:204]: Just to be sure, `$backup` when used here will always have relevant data properly adjusted already ?

**Review by sonicaj — APPROVED** [review]
Are we actually changing the format of questions atm as well because that will result in a decent number of migrations to be written ?

File: .github/ct-install-config/common-ct-lint.yaml (hunk 1 above)
Comment (stavros-k) [reply .github/ct-install-config/common-ct-lint.yaml:3]: Was never valid option, just didnt throw an error before!

File: library/common/templates/app_functions/_postgres.tpl (hunk 2 above)
Comment (stavros-k) [reply library/common/templates/app_functions/_postgres.tpl:204]: Yes! there is a sanity check few lines above to check the dictionary is not empty,
And if there are any required values missing, common chart will throw an error too

**Comment by stavros-k** [comment]
> Are we actually changing the format of questions atm as well because that will result in a decent number of migrations to be written ?

Yes, with the current format, it is impossible to add ACLs.

I've created a mostly standardized migration here: 
https://github.com/truenas/charts/pull/1688/files#diff-5ddac63049175241bcba93821613b955668822fa67738f7ae4257ea1bca09ac0

**Pull request merged**